Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp

```diff
@@ -482,6 +489,8 @@ void Recorder::strokePath(const Path& path)
             recordStrokeBezierCurve(path.inlineData<BezierCurveData>());
         return;
     }
+#else
+    appendStateChangeItemIfNecessary();
```

Comment:
And this one could be moved just before 

```
recordStrokePath(path);
```

So it is clearer that every time we record a drawing command, we are sure the state was already updated.